plugins/parking.py

```diff
@@ -1,76 +1,44 @@
-        pass
-
-    def registerCommand(self, irc):
-        irc.registerCommand('!parking', 'Parken')
-
-    def _get_parking_status(self):
-        url = BYTEBOT_PLUGIN_CONFIG['parking']['url']
-
-        data = urllib2.urlopen(url, timeout=BYTEBOT_HTTP_TIMEOUT).read(
-            BYTEBOT_HTTP_MAXSIZE)
-
-        data = unicode(data, errors='ignore')
-
-        ret = json.loads(data)
-
-        return ret
-
-    def onPrivmsg(self, irc, msg, channel, user):
-        if msg.find('!parking') == -1:
-            return
-
-        self.irc = irc
-        self.channel = channel
-
-        try:
-            last_parking = irc.last_parking
-        except Exception as e:
-            last_parking = 0
-
-        if last_parking < (time() - 60):
-            try:
-                data = self._get_parking_status()
-
-                irc.msg(channel, 'Free parking lots:')
-
-                for x in range(1, len(data)):
-
-                    name = data[x][u'name'].encode('ascii', 'ignore')
-                    occupied = int(data[x][u'belegt'].encode('ascii',
-                                                             'ignore'))
-                    spaces = int(data[x][u'maximal'].encode('ascii', 'ignore'))
-
-                    if(occupied < 0):
-                        occupied = 0
-
-                    if(spaces <= 0):
-                        print_str = '{:25s}: not available'.format(name)
-                    else:
-                        print_str = '{:25s}: '.format(name) + \
-                                    '{:3.0f} / '.format(spaces - occupied) + \
-                                    '{:3.0f}'.format(spaces)
-
-                    irc.msg(channel, print_str)
-
-                irc.last_parking = time()
-
-            except Exception as e:
-                print(e)
-                irc.msg(channel, 'Error while fetching data.')
-        else:
-            irc.msg(channel, "Don't overdo it ;)")
+from irc3 import asyncio
+import aiohttp
+import xml.etree.ElementTree as ET
+
+
+@command(permission="view")
+@asyncio.coroutine
+def parking(bot, mask, target, args):
+    """Show the current parking lot status
+
+        %%parking
+    """
+    config = BYTEBOT_PLUGIN_CONFIG['parking']
+
+    if config['url'] == "parking_url":
+        return "I don't have your parking url!"
+
+    bot.privmsg(target, 'Parkhausbelegung:')
+
+    with aiohttp.Timeout(10):
+        with aiohttp.ClientSession(loop=bot.loop) as session:
+            resp = yield from session.get(config['url'])
+            if resp.status != 200:
+                bot.privmsg(target, "Error while retrieving parking data")
+                raise Exception()
+            r = yield from resp.read()
+
+            root = ET.fromstring(r)
+            for lot in root.findall('ph'):
+                name = lot.find('longname').text
+                use = int(lot.find('belegung').text)
+                max = int(lot.find('kapazitaet').text)
+
+                print_str = \
```

Comment:
The variable can be reduced - it's only used in the privmsg call. It could also be nice to simply use one single string with multiple parameters passed to format, that should (in theory) look less cluttered.